src/sbt-slicer.cpp

```diff
@@ -1,156 +1,141 @@
+        "dump-dg-only",
+        llvm::cl::desc("Only dump dependence graph to dot,"
+                       " do not slice the module (default=false)."),
+        llvm::cl::init(false), llvm::cl::cat(SlicingOpts));
+
+llvm::cl::opt<bool> dump_bb_only(
+        "dump-bb-only",
+        llvm::cl::desc("Only dump basic blocks of dependence graph to dot"
+                       " (default=false)."),
+        llvm::cl::init(false), llvm::cl::cat(SlicingOpts));
+
+llvm::cl::opt<bool> criteria_are_next_instr(
+        "criteria-are-next-instr",
+        llvm::cl::desc(
+                "Assume that slicing criteria are not the call-sites\n"
+                "of the given function, but the instructions that\n"
+                "follow the call. I.e. the call is used just to mark\n"
+                "the instruction.\n"
+                "E.g. for 'crit' being set as the criterion, slicing critera "
+                "are all instructions that follow any call of 'crit'.\n"),
+        llvm::cl::init(false), llvm::cl::cat(SlicingOpts));
+
+llvm::cl::opt<std::string> annotationOpts(
+        "annotate",
+        llvm::cl::desc(
+                "Save annotated version of module as a text (.ll).\n"
+                "Options:\n"
+                "  dd: data dependencies,\n"
+                "  cd:control dependencies,\n"
+                "  pta: points-to information,\n"
+                "  memacc: memory accesses of instructions,\n"
+                "  slice: comment out what is going to be sliced away).\n"
+                "for more options, use comma separated list"),
+        llvm::cl::value_desc("val1,val2,..."), llvm::cl::init(""),
+        llvm::cl::cat(SlicingOpts));
+
+llvm::cl::opt<bool> memsafety(
+        "memsafety",
+        llvm::cl::desc(
+                "Assume that slicing criteria are potentially memory-unsafe "
+                "instructions. This option implies criteria-are-next-instr, "
+                "i.e., the slicing criteria are \"only marked\" "
+                "by the given slicing criteria."),
+        llvm::cl::init(false), llvm::cl::cat(SlicingOpts));
+
+llvm::cl::opt<bool> cutoff_diverging(
+        "cutoff-diverging",
+        llvm::cl::desc(
+                "Cutoff diverging paths. That is, call abort() on those paths "
+                "that may not reach the slicing criterion "
+                " (default=true)."),
+        llvm::cl::init(true), llvm::cl::cat(SlicingOpts));
 
 // mapping of AllocaInst to the names of C variables
 std::map<const llvm::Value *, std::string> valuesToVariables;
 
-static void maybe_print_statistics(llvm::Module *M, const char *prefix = nullptr)
-{
+static void maybe_print_statistics(llvm::Module *M,
+                                   const char *prefix = nullptr) {
     if (!statistics)
         return;
 
     using namespace llvm;
     uint64_t inum, bnum, fnum, gnum;
     inum = bnum = fnum = gnum = 0;
 
-    for (const Function& F : *M) {
+    for (auto &I : *M) {
```

Comment:
I know that this name change is completely wrong (Module consists of Functions) but the aim of this PR is to minimise differences between this project and DG as much as possible.

Reply:
Wouldn't it be better to file a PR in DG that fixes the name there?

Reply:
It definitely would. PR: https://github.com/mchalupa/dg/pull/440t